cdk/l3-cdk-resources/atlas-private-endpoint/src/integ.default.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+import * as cdk from 'aws-cdk-lib';
+import {ApiKeyDefinition, AtlasBasicProps} from "@mongodbatlas-awscdk/atlas-basic";
+import {AtlasBasicPrivateEndpoint, PrivateEndpointProps} from "./index";
+
+const app = new cdk.App();
+
+const stack = new cdk.Stack(app, 'atlas-basic-default', {
+    env: { region: process.env.CDK_DEFAULT_REGION, account: process.env.CDK_DEFAULT_ACCOUNT },
+});
+
+const apiKeys: ApiKeyDefinition = {
+    privateKey: stack.node.tryGetContext('MONGODB_ATLAS_PRIVATE_KEY') || process.env.MONGODB_ATLAS_PRIVATE_KEY,
+    publicKey: stack.node.tryGetContext('MONGODB_ATLAS_PUBLIC_KEY') || process.env.MONGODB_ATLAS_PUBLIC_KEY,
+};
+
+const orgId = stack.node.tryGetContext('MONGODB_ATLAS_ORG_ID') || process.env.MONGODB_ATLAS_ORG_ID;
+const vpcId = stack.node.tryGetContext('AWS_VPC_ID') || process.env.AWS_VPC_ID;
+const subnetId = stack.node.tryGetContext('AWS_SUBNET_ID') || process.env.AWS_SUBNET_ID;
+const awsRegion = stack.node.tryGetContext('AWS_REGION') || process.env.AWS_REGION;
+
+const replicationSpecs = [
+    {
+        numShards: 1,
+        advancedRegionConfigs: [
+            {
+                analyticsSpecs: {
+                    ebsVolumeType: 'STANDARD',
+                    instanceSize: 'M10',
+                    nodeCount: 1,
+                },
+                electableSpecs: {
+                    ebsVolumeType: 'STANDARD',
+                    instanceSize: 'M10',
+                    nodeCount: 3,
+                },
+                priority: 7,
+                regionName: 'US_EAST_1',
+            },
+        ],
+    },
+];
+
+const atlasBasicProps : AtlasBasicProps = {
+    apiKeys: apiKeys,
+    clusterProps: {
+        replicationSpecs: replicationSpecs,
+    },
+    projectProps: {
+        orgId: orgId,
+    },
+}
+
+const privateEndpointProps : PrivateEndpointProps = {
+    privateEndpoints: [{
+        vpcId: vpcId,
+        subnetIds: [subnetId]
+    }],
+}
+
+const props   = {
+    apiKeys: apiKeys,
+    atlasBasicProps: atlasBasicProps,
+    privateEndpointProps: privateEndpointProps,
+    region: awsRegion
+}
```

Comment:
[q] Can we just define a single interface with all the properties instead of asking the user to create 3 different interfaces?

Reply:
[a] props is a single interface. The user is free to populate it as he likes. It is an app code.